lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -1676,7 +1696,14 @@ private function getSelectConditionStatementColumnSQL($field, $assoc = null)
                 ? $this->class->fieldMappings[$field]['inherited']
                 : $this->class->name;
 
-            return [$this->getSQLTableAlias($className) . '.' . $this->quoteStrategy->getColumnName($field, $this->class, $this->platform)];
+            // Fix for bug GH-8229 (id column from parent class renamed in child class):
+            // Use the correct name for the id columns as named in the inherited class.
+            $class = $this->class;
```

Comment:
WDYT?
```suggestion
            $nonInheritedClass = $this->class;
            if ($className !== $nonInheritedClass->name) {
```

Doing `$className !== $this->class->name` is kinda confusing

Reply:
You're right, that's confusing. I will optimize this part and merge it with the condition in the lines above.